Crashlytics/Crashlytics/Models/FIRCLSFileManager.m

```diff
@@ -304,27 +306,14 @@ - (BOOL)removeContentsOfPendingPath {
 
 - (BOOL)removeContentsOfAllPaths {
   BOOL success = YES;
-
-  // We want to call all of these methods, even if some fail, and return
-  // NO if any fail. This turned out to be slightly tricky to do in more
-  // compact forms, so I did it the simple but verbose way.
-
-  if (![self removeContentsOfProcessingPath]) {
-    success = NO;
-  }
-
-  if (![self removeContentsOfPendingPath]) {
-    success = NO;
-  }
-
-  if (![self removeContentsOfDirectoryAtPath:self.preparedPath]) {
-    success = NO;
-  }
-
-  if (![self removeContentsOfDirectoryAtPath:self.activePath]) {
-    success = NO;
-  }
-
+  BOOL contentsOfProcessingPathRemoved = [self removeContentsOfProcessingPath];
+  BOOL contentsOfPendingPathRemoved = [self removeContentsOfPendingPath];
+  BOOL contentsOfDirectoryAtPreparedPathRemoved =
+      [self removeContentsOfDirectoryAtPath:self.preparedPath];
+  BOOL contentsOfDirectoryAtActivePathRemoved =
+      [self removeContentsOfDirectoryAtPath:self.activePath];
+  success = contentsOfProcessingPathRemoved && contentsOfPendingPathRemoved &&
+            contentsOfDirectoryAtPreparedPathRemoved && contentsOfDirectoryAtActivePathRemoved;
```

Comment:
Here might be this type of form of code if you wish:
success = [self removeContentsOfProcessingPath] & [self removeContentsOfPendingPath] & ... etc
Imo above case is more friendly for debug =)

Reply:
We can declare success at the time we check for all the criteria:
BOOL success = contentsOfProcessingPathRemoved && contentsOfPendingPathRemoved &&
	            contentsOfDirectoryAtPreparedPathRemoved && contentsOfDirectoryAtActivePathRemoved;

Reply:
> We can declare success at the time we check for all the criteria:
> BOOL success = contentsOfProcessingPathRemoved && contentsOfPendingPathRemoved &&
> contentsOfDirectoryAtPreparedPathRemoved && contentsOfDirectoryAtActivePathRemoved;

You are definitely right :)
Thanks!